qiskit_experiments/calibration_management/base_calibration_experiment.py

```diff
@@ -0,0 +1,338 @@
+                    pulse.Gaussian(
+                        duration=self.experiment_options.duration,
+                        amp=Parameter("amp"),
+                        sigma=self.experiment_options.sigma,
+                    ),
+                    pulse.DriveChannel(self.physical_qubits[0]),
+            )
+
+        """
+        raise NotImplementedError(
+            f"{self.__class__.__name__} could not find a schedule in the experiment options "
+            "or the calibrations and no default schedule method was implemented."
+        )
+
+    def _validate_schedule(self, schedule: ScheduleBlock):
+        """Subclass can implement this method to validate the schedule they use.
+
+        Validating schedules may include checks on the number of parameters and
+        the channels in the schedule. The functions :meth:`_validate_channels` and
+        :meth:`_validate_parameters` implement such standard checks for reuse.
+        """
+
+    def _validate_channels(self, schedule: ScheduleBlock):
+        """Check that the physical qubits are contained in the schedule.
+
+        This is a helper method that experiment developers can call in their implementation
+        of :meth:`validate_schedules` when checking the schedules.
+
+        Args:
+            schedule: The schedule for which to check the qubits.
+
+        Raises:
+            CalibrationError: If a physical qubit is not contained in the channels schedule.
+        """
+        for qubit in self.physical_qubits:
+            if qubit not in set(ch.index for ch in schedule.channels):
+                raise CalibrationError(
+                    f"Schedule {schedule.name} does not contain a channel "
+                    f"for the physical qubit {qubit}."
+                )
+
+    def _validate_parameters(self, schedule: ScheduleBlock, n_expected_parameters: int):
+        """Check that the schedule has the expected number of parameters.
+
+        This is a helper method that experiment developers can call in their implementation
+        of :meth:`validate_schedules` when checking the schedules.
+
+        Args:
+            schedule: The schedule for which to check the qubits.
+            n_expected_parameters: The number of free parameters the schedule must have.
+
+        Raises:
+            CalibrationError: If the schedule does not have n_expected_parameters parameters.
+        """
+        if len(schedule.parameters) != n_expected_parameters:
+            raise CalibrationError(
+                f"The schedules {schedule.name} for {self.__class__.__name__} must have "
+                f"{n_expected_parameters} parameters. Found {len(schedule.parameters)}."
+            )
+
+    def get_schedule(
+        self,
+        qubits: Optional[Tuple[int, ...]] = None,
+        sched_name: Optional[str] = None,
+        option_name: str = "schedule",
+        assign_params: Optional[Dict[str, Parameter]] = None,
+        **kwargs,
+    ) -> ScheduleBlock:
```

Comment:
The `get_schedule` and relevant private methods should be really in the mixin? For example, we may want to run some characterization with calibration, and want to avoid parameter update. For example, `QubitSpectroscopy` may want to write `_get_schedule_from_options`. Probably we want `CalibrationMixIn` and `PulseGateMixIn`? This could harm readability though.

Reply:
Having several MixIns is too much. It's already not the most straightforward class Hierarchy as is.